### ISSUE TITLE
`--clear` now clears the screen even when _not_ restarting

### ISSUE THREAD

**Issue by max-sixty** [watchexec/watchexec#809]
I use both `-c` and `-rc`. `-rc` restarts on any change; `-c` only restarts on a change when exiting.

On previous versions of watchexec, both would clear the screen when they restarted.

On the latest version, both clear the screen on any change. So `-c` will clear even when not restarting. Is that intentional?

```
watchexec 1.25.1 (2024-04-07) +pid1
build-date: 2024-04-07
release: 1.25.1
features: default,pid1
Mac OS
```

Simple repro:

```
watchexec -c -- '
   while true
   do
   echo "Hello, World!"
   echo "World, Hi!"
   sleep 0.1
   done
   '
```

...now make a change and the screen should clear. (if it's just me, lmk and I'll make a full log file etc)

Thank you!

**Comment by passcod** [comment]
I think that's intentional, yes. As I conceptualise it, `-c` (with or without any other flags) should clear every time the command starts, just before it starts, regardless of why it's starting (or restarting).

I hadn't realised this wasn't what it was doing previously and I think I would consider the previous behaviour a bug then. Do you have a usecase for the previous behaviour?

**Comment by max-sixty** [comment]
> I think that's intentional, yes. As I conceptualise it, `-c` (with or without any other flags) should clear every time the command starts, just before it starts, regardless of why it's starting (or restarting).

Totally! But the new behavior restarts on any change _even when that change doesn't result in a restart_. Does that make sense? Sorry if the original was unclear...

**Comment by passcod** [comment]
Oh yeah I see what you mean. Hmm. That's definitely a bug, I think introduced due to another bugfix where `--print-events` would action before the clear.

**Comment by passcod** [comment]
Will be fixed in 2.0.0

**Comment by max-sixty** [comment]
Thanks @passcod !
